### PR TITLE
Deb packaging

### PR DESCRIPTION
**Pull request by dmth**
Updated Packaging.

1. Require imbox 0.8 as 0.6 is buggy when receiving multipart e-mails with attachments
2. Add Script to update data for the asn-lookup-expert, maxmind-geoip and tor-nodes-expert with cron

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -37,6 +37,12 @@
     ('/opt/intelmq/var/lib/bots/file-output/',
      [],
      ),
+    ('/usr/bin',
```
Comment (sebix) [inline setup.py]: Why not using `/usr/local/bin` here too?

**Comment by sebix** [comment]
We already home some cronjob scripts updating data: https://github.com/certtools/intelmq/tree/master/contrib/cron-jobs

Could these be brought into sync?

File: setup.py (hunk 1 above)
Comment (dmth) [inline setup.py]: It is bad practise to install something to `/usr/local/` from a package. This Folder is intended for manual "installations" w/o packages.

**Comment by dmth** [comment]
We have not been aware of these scripts. Thank you for pointing them out.

After a quick glance, we advise to use the scripts provided by us.
They are more robust. Maybe just link from contrib/cron-jobs to the respective bots directory?

File: intelmq/bots/experts/tor_nodes/update_tor_nodes.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+# Update tor nodes data
+PATH=/bin:/usr/bin
+
+set -e
+trap cleanup EXIT
+
+DEST_FILE=/opt/intelmq/var/lib/bots/tor_nodes/tor_nodes.dat
```
Comment (aaronkaplan) [inline intelmq/bots/experts/tor_nodes/update_tor_nodes.sh]: hardcoded path! Thin might not be true if switching to LSB

File: intelmq/bots/experts/tor_nodes/update_tor_nodes.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+{
+  [ -e "$TMP_DIR/latest.bz2" ] && rm "$TMP_DIR/latest.bz2"
+  rmdir "$TMP_DIR"
+}
+
+
+[ -d "$dest_dir" ] || mkdir -p "$dest_dir"
+
+cd "$TMP_DIR"
+curl -O https://internet2.us/static/latest.bz2
```
Comment (aaronkaplan) [inline intelmq/bots/experts/tor_nodes/update_tor_nodes.sh]: what happens when this fails (for example, the server is not reachable/down)?

File: intelmq/bots/experts/tor_nodes/update_tor_nodes.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+cleanup()
+{
+  [ -e "$TMP_DIR/latest.bz2" ] && rm "$TMP_DIR/latest.bz2"
+  rmdir "$TMP_DIR"
+}
+
+
+[ -d "$dest_dir" ] || mkdir -p "$dest_dir"
+
+cd "$TMP_DIR"
```
Comment (aaronkaplan) [inline intelmq/bots/experts/tor_nodes/update_tor_nodes.sh]: does $TMP_DIR always have enough space?

File: intelmq/bots/experts/asn_lookup/update_asn_data.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+{
+  [ -d "$dest_dir" ] && rm -rf "$TMP_DIR"
+}
+
+fetch_and_install()
+{
+  cd "$TMP_DIR"
+  # FIXME: we have to use http as the data is not available via httpd,
+  # so this is not a secure connection... :-(
+  curl -k -s -O http://data.ris.ripe.net/rrc12/latest-bview.gz
```
Comment (aaronkaplan) [inline intelmq/bots/experts/asn_lookup/update_asn_data.sh]: The data from there will not work with pyasn 1.5. See #516

File: intelmq/bots/experts/maxmind_geoip/update_geoip_data.sh [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+{
+  [ -d "$dest_dir" ] && rm -rf "$TMP_DIR"
+}
+
+fetch_and_install()
+{
+  cd "$TMP_DIR"
+  curl -s -O https://geolite.maxmind.com/download/geoip/database/GeoLite2-City.mmdb.gz
+  gunzip GeoLite2-City.mmdb
+  mv GeoLite2-City.mmdb "$DEST_FILE"
```
Comment (aaronkaplan) [inline intelmq/bots/experts/maxmind_geoip/update_geoip_data.sh]: missing: HUP signal for the bot in order to reload

File: intelmq/bots/experts/asn_lookup/update_asn_data.sh [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  gunzip -c <latest-bview.gz | bzip2 -c >latest-bview.bz2 
+  pyasn_util_convert.py --single --no-progress latest-bview.bz2 ipasn.dat
+  mv ipasn.dat "$DEST_FILE"
```
Comment (aaronkaplan) [inline intelmq/bots/experts/asn_lookup/update_asn_data.sh]: missing: HUP signal for the bot in order to reload

File: intelmq/bots/experts/tor_nodes/update_tor_nodes.sh [hunk 8]
```diff
@@ -1,24 +1,37 @@
...
+  [ -d "$dest_dir" ] && rm -rf "$TMP_DIR"
+}
 
-[ -d "$dest_dir" ] || mkdir -p "$dest_dir"
+fetch_and_install()
+{
+  cd "$TMP_DIR"
+  curl -s -O https://internet2.us/static/latest.bz2
+  bunzip2 -c latest.bz2 >tor_nodes.dat
+  mv tor_nodes.dat "$DEST_FILE"
```
Comment (aaronkaplan) [inline intelmq/bots/experts/tor_nodes/update_tor_nodes.sh]: missing: HUP signal for the bot in order to reload

**Comment by aaronkaplan** [comment]
this is really a duplicate of the cron scripts we already have in contrib/
Please work your scripts into the existing ones and compare functionality.

File: intelmq/bots/experts/maxmind_geoip/update_geoip_data.sh (hunk 6 above)
Comment (sebix) [inline intelmq/bots/experts/maxmind_geoip/update_geoip_data.sh]: #508 is not merged yet, so how should he have known?

File: intelmq/bots/experts/asn_lookup/update-asn-data [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+{
+  [ -d "$dest_dir" ] && rm -rf "$TMP_DIR"
+}
+
+fetch_and_install()
+{
+  cd "$TMP_DIR"
+  # FIXME: we have to use http as the data is not available via httpd,
+  # so this is not a secure connection... :-(
+  curl -k -s -O http://data.ris.ripe.net/rrc12/latest-bview.gz
```
Comment (aaronkaplan) [inline intelmq/bots/experts/asn_lookup/update-asn-data]: Missing http proxy settings here.

File: intelmq/bots/experts/maxmind_geoip/update-geoip-data [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+cleanup()
+{
+  [ -d "$dest_dir" ] && rm -rf "$TMP_DIR"
+}
+
+fetch_and_install()
+{
+  cd "$TMP_DIR"
+  curl -s -O https://geolite.maxmind.com/download/geoip/database/GeoLite2-City.mmdb.gz
```
Comment (aaronkaplan) [inline intelmq/bots/experts/maxmind_geoip/update-geoip-data]: http proxy

File: intelmq/bots/experts/tor_nodes/update-tor-nodes [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+cleanup()
+{
+  [ -d "$dest_dir" ] && rm -rf "$TMP_DIR"
+}
+
+fetch_and_install()
+{
+  cd "$TMP_DIR"
+  curl -s -O https://internet2.us/static/latest.bz2
```
Comment (aaronkaplan) [inline intelmq/bots/experts/tor_nodes/update-tor-nodes]: http proxy

**Comment by aaronkaplan** [comment]
okay, I re-reviewed it. Going to pull this PR as soon as the curl is in --silent mode
(same for the other tools)

File: setup.py (hunk 1 above)
Comment (sebix) [inline setup.py]: Just noticed that this affects all installations, not only the packaged ones. These scripts should go to the [`console_scripts`](https://github.com/certtools/intelmq/blob/master/setup.py#L96), not "hard"copied to `/usr/bin`. They may be installed there, if the path for binaries is set to `/usr/bin`, which is the case for setuptools used for packaging, but not for locally installed instances.

**Pull request merged**